fall back to /usr/share/defaults/etc/hosts if no such file exists in /etc


--- glibc-2.25/nss/nss_files/files-XXX.c~	2017-02-05 15:28:43.000000000 +0000
+++ glibc-2.25/nss/nss_files/files-XXX.c	2017-07-01 14:58:52.184575322 +0000
@@ -39,6 +39,7 @@
 #define ENTNAME_r	CONCAT(ENTNAME,_r)
 
 #define DATAFILE	"/etc/" DATABASE
+#define ALTDATAFILE	"/usr/share/defaults/etc/" DATABASE
 
 #ifdef NEED_H_ERRNO
 # include <netdb.h>
@@ -77,6 +78,9 @@
       *stream = __nss_files_fopen (DATAFILE);
 
       if (*stream == NULL)
+        *stream = __nss_files_fopen (ALTDATAFILE);
+
+      if (*stream == NULL)
 	status = errno == EAGAIN ? NSS_STATUS_TRYAGAIN : NSS_STATUS_UNAVAIL;
     }
   else
